lib/ddtrace/contrib/ethon/easy_patch.rb

```diff
@@ -0,0 +1,136 @@
+        module InstanceMethods
+          def http_request(url, action_name, options = {})
+            return super unless tracer_enabled?
+
+            # It's tricky to get HTTP method from libcurl
+            @datadog_method = action_name.to_s.upcase
+            super
+          end
+
+          def headers=(headers)
+            return super unless tracer_enabled?
+
+            # Store headers to call this method again when span is ready
+            @datadog_original_headers = headers
+            super headers
+          end
+
+          def perform
+            return super unless tracer_enabled?
+            datadog_before_request
+            super
+          end
+
+          def complete
+            return super unless tracer_enabled?
+            begin
+              response_options = mirror.options
+              response_code = (response_options[:response_code] || response_options[:code]).to_i
+              if response_code.zero?
+                return_code = response_options[:return_code]
+                message = return_code ? ::Ethon::Curl.easy_strerror(return_code) : 'unknown reason'
+                set_span_error_message("Request has failed: #{message}")
+              else
+                @datadog_span.set_tag(Datadog::Ext::HTTP::STATUS_CODE, response_code)
+                if Datadog::Ext::HTTP::ERROR_RANGE.cover?(response_code)
+                  set_span_error_message("Request has failed with HTTP error: #{response_code}")
+                end
+              end
+            ensure
+              @datadog_span.finish
+              @datadog_span = nil
+            end
+            super
+          end
+
+          def reset
+            super
+          ensure
+            if tracer_enabled?
+              @datadog_span = nil
+              @datadog_method = nil
+              @datadog_original_headers = nil
+            end
+          end
+
+          def datadog_before_request(parent_span: nil)
+            @datadog_span = datadog_configuration[:tracer].trace(
+              Ext::SPAN_REQUEST,
+              service: datadog_configuration[:service_name],
+              span_type: Datadog::Ext::HTTP::TYPE_OUTBOUND
+            )
+            @datadog_span.parent = parent_span unless parent_span.nil?
+
+            datadog_tag_request
+
+            if datadog_configuration[:distributed_tracing]
+              @datadog_original_headers ||= {}
+              Datadog::HTTPPropagator.inject!(@datadog_span.context, @datadog_original_headers)
+              self.headers = @datadog_original_headers
+            end
+          end
+
+          def datadog_span_started?
+            instance_variable_defined?(:@datadog_span) && !@datadog_span.nil?
+          end
+
+          private
+
+          def datadog_tag_request
+            span = @datadog_span
+            uri = URI.parse(url)
+            method = defined?(@datadog_method) ? @datadog_method.to_s : ''
+            span.resource = "#{method} #{uri.path}".lstrip
```

Comment:
We only want to set the HTTP method here, e.g. `GET`, but omit the path.

Sounds crazy, and I see the value of including the path. However, the resource is meant to be a `GROUP BY` like key, which in practice requires values that meaningfully group traces on some user-defined dimension. HTTP paths often contain unique input (such as numeric IDs, API tokens), which tend to flatten these traces into groups too small to be meaningful. Hence, as a default behavior, we've been only setting the method as the resource for HTTP client integrations.

If the omission of `path` as a default behavior is an issue for users, the concession we could offer is the option to customize their span and set their own resource using a callback. See HTTP as an example: https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/contrib/http/instrumentation.rb#L80. Implementing something like this is optional though, not necessary to merge this PR.

Also, it looks like the `method` can be `nil`; if you remove the path, it looks like the resource could then be an empty string, which will cause the trace to be malformed and dropped. We'll want to make sure resource is never blank, even if we have to give it some kind of default value instead.

Reply:
Interesting, I was wondering why the resource contains only method name in other instrumentations. I changed the default value of `method` to `N/A` & handled the case when it is `nil`.